### ISSUE TITLE
[Bug]Crash when import Linus's key

### ISSUE THREAD

**Issue by Tedko** [DimensionDev/Tessercube-iOS#27]
「出问题的那个key」：https://keyserver.ubuntu.com/pks/lookup?op=get&search=0xabaf11c65a2970b130abe3c479be3e4300411886

Quote from user: 
>可能是因为多个电子邮件id

[TesserCube-2019-07-31-003524.txt](https://github.com/DimensionDev/Tessercube-iOS/files/3448477/TesserCube-2019-07-31-003524.txt)
[TesserCube-2019-07-31-003450.txt](https://github.com/DimensionDev/Tessercube-iOS/files/3448480/TesserCube-2019-07-31-003450.txt)

**Comment by jk234ert** [comment]
在最新版本上导入，没发现问题啊

[![IMG_0158.png](https://i.loli.net/2019/07/31/5d40ee336328759735.png)](https://i.loli.net/2019/07/31/5d40ee336328759735.png)
[![IMG_0159.png](https://i.loli.net/2019/07/31/5d40ee3363e1b86296.png)](https://i.loli.net/2019/07/31/5d40ee3363e1b86296.png)
[![IMG_0157.png](https://i.loli.net/2019/07/31/5d40ee3362a0b53903.png)](https://i.loli.net/2019/07/31/5d40ee3362a0b53903.png)

**Comment by Tedko** [comment]
看看报错文件？

**Comment by Tedko** [comment]
@imlibra

**Comment by MainasuK** [comment]
@jk234ert 
Delete two contacts after import that key.
Then re-import will crash the App at `MeViewModel.tableView(_:cellForRowAt:)`.

It is looks like the related issue #18

**Comment by jk234ert** [comment]
> @jk234ert
> Delete two contacts after import that key.
> Then re-import will crash the App at `MeViewModel.tableView(_:cellForRowAt:)`.
> 
> It is looks like the related issue #18

Diagnosing...

**Comment by MainasuK** [comment]
![image](https://user-images.githubusercontent.com/7940186/62178934-53386700-b37d-11e9-9295-c8d38ef23660.png)
Import and remove keys several times make sure the Me tab display empty rows.
(it's should be placeholder cell but somehow not).

Console
```
Fatal error: Index out of range
2019-07-31 10:23:00.066097+0800 TesserCube[22129:13640961] Fatal error: Index out of range
```

Stack
![image](https://user-images.githubusercontent.com/7940186/62179061-c04bfc80-b37d-11e9-8f25-6e2f308307c7.png)

**Comment by jk234ert** [comment]
fix pushed in pr #28

**Comment by Tedko** [comment]
z👍

**Comment by MainasuK** [comment]
This bug should be resolved in 22b0f08d37e256cc3241be2e26265e9a0df0aa6e (finally).

ViewModel drive hasKey emit next to reload actions view. Then the `view.setNeedsLayout()` cause table view reload but at that time data source also changed. The index path point to one deleted element.
 
```swift
// MeViewModel
hasKey = keys.asDriver().map { !$0.isEmpty }

// MeViewController
viewModel.hasKey
    .drive(onNext: { [weak self] result in
        self?.reloadActionsView(hasKey: result)
    })
    .disposed(by: disposeBag)
```

**Comment by Tedko** [comment]
z👍
